### DISCUSSION TITLE
Best startup time with Logback-tyler

### DISCUSSION THREAD

**Discussion by Dichotomia** [qos-ch/logback-tyler#4] [Q&A] [answered]
Hello.
I have some difficulties to find information about this project.
Is it possible to startup faster the java application with logback-tyler instead of XML configuration?

Best regards

**Comment by ceki** [comment, marked as answer]
@Dichotomia A `TylerConfigurator` will start faster since there is no need to parse an XML file and process it. 

With recent versions of logback and logback-tyler you can still configure logger levels dynamically using [properties files](https://logback.qos.ch/manual/configuration.html#propertiesConfigurator). Note that configuration files in properties format can be watched for changes. Configuration files in properties format are special in the sense that they can be watched for changes on web-servers using HTTP or HTTPS protocols. For security reasons,  XML configuration files cannot be loaded via HTTP or HTTPS.

**Reply by Dichotomia** [reply]
Thanks @ceki for that fast answer. That's look promising.
Do you have any example of application using Java Configuration instead of XML configuration. And what is the use of logback-tyler in this case? I found some information but no practical tutorial.

Best regards

**Comment by ceki** [comment]
@Dichotomia There are many examples in the logback manual. Whenever you see an example configuration file, just click on the tab labeled "Tytler". You can also tylerize your own configuration files using the [translation services](https://logback.qos.ch/translator/).

Once you generate the configurator in Java (using logback-tyler), the generated file contains further instructions on how to integrate your custom configurator in Java into your project. It is actually quite simple. 

Keep in mind that by integrating a .properties configuration file info your configurator, you can still change logger levels dynamically, without redeploying your application.

**Comment by Dichotomia** [comment]
The questions have been answered
